src/lightning/data/streaming/shuffle.py

```diff
@@ -24,17 +24,27 @@
 class Shuffle(ABC):
     """Shuffle describe how to distribute chunked datasets across processes and workers."""
 
-    def __init__(self, cache: Cache, seed: int):
+    def __init__(self, cache: Cache, seed: int, drop_last: bool):
         self.cache = cache
         self.seed = seed
+        self.drop_last = drop_last
         self.random_state = None
 
-    @abstractmethod
+    @lru_cache(maxsize=10)
     def get_len(self, distributed_env: _DistributedEnv, current_epoch: int) -> int:
-        pass
+        _, intervals_per_ranks = self.get_chunks_and_intervals_per_ranks(distributed_env, current_epoch)
+
+        if self.drop_last:
+            items_per_process = [
+                sum([(interval[-1] - interval[0]) for interval in intervals]) for intervals in intervals_per_ranks
+            ]
+            min_items_per_process = min(items_per_process)
+            return min_items_per_process
+
+        return sum([(interval[-1] - interval[0]) for interval in intervals_per_ranks[distributed_env.global_rank]])
```

Comment:
```suggestion
        return sum((interval[-1] - interval[0]) for interval in intervals_per_ranks[distributed_env.global_rank])
```
sum can directly work with the generator